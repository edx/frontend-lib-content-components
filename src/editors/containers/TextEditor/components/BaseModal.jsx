import React from 'react';
import PropTypes from 'prop-types';

import {
  ActionRow,
  ModalDialog,
} from '@edx/paragon';

export const BaseModal = ({
  isOpen,
  close,
  title,
  children,
  confirmAction,
  footerAction,
}) => (
  <ModalDialog
    title="My dialog"
    isOpen={isOpen}
    onClose={close}
    size="lg"
    variant="default"
    hasCloseButton
    isFullscreenOnMobile
    isFullscreenScroll
  >
    <ModalDialog.Header>
      <ModalDialog.Title>
        {title}
      </ModalDialog.Title>
    </ModalDialog.Header>
    <ModalDialog.Body>
      {children}
    </ModalDialog.Body>
    <ModalDialog.Footer>
      <ActionRow>
        {footerAction}
        <ActionRow.Spacer />
        <ModalDialog.CloseButton variant="tertiary" onClick={close}>
          Cancel
        </ModalDialog.CloseButton>
        {confirmAction}
      </ActionRow>
    </ModalDialog.Footer>
  </ModalDialog>
);

BaseModal.defaultProps = {
  footerAction: null,
};

BaseModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  close: PropTypes.func.isRequired,
  title: PropTypes.node.isRequired,
  children: PropTypes.node.isRequired,
  confirmAction: PropTypes.node.isRequired,
  footerAction: PropTypes.node,
};

export default BaseModal;
