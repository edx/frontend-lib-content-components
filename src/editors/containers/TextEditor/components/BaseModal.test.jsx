import React from 'react';
import { shallow } from 'enzyme';

import { BaseModal } from './BaseModal';

describe('BaseModal ImageUploadModal template component', () => {
  test('snapshot', () => {
    const props = {
      isOpen: true,
      close: jest.fn().mockName('props.close'),
      title: 'props.title node',
      children: 'props.children node',
      confirmAction: 'props.confirmAction node',
      footerAction: 'props.footerAction node',
    };
    expect(shallow(<BaseModal {...props} />)).toMatchSnapshot();
  });
});
