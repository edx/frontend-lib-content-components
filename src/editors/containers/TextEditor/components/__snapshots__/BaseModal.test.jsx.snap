// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseModal ImageUploadModal template component snapshot 1`] = `
<ModalDialog
  hasCloseButton={true}
  isFullscreenOnMobile={true}
  isOpen={true}
  onClose={[MockFunction props.close]}
  size="lg"
  title="My dialog"
  variant="default"
>
  <ModalDialog.Header>
    <ModalDialog.Title>
      props.title node
    </ModalDialog.Title>
  </ModalDialog.Header>
  <ModalDialog.Body>
    props.children node
  </ModalDialog.Body>
  <ModalDialog.Footer>
    <ActionRow>
      <ModalDialog.CloseButton
        onClick={[MockFunction props.close]}
        variant="tertiary"
      >
        Cancel
      </ModalDialog.CloseButton>
      props.confirmAction node
    </ActionRow>
  </ModalDialog.Footer>
</ModalDialog>
`;
